### PR TITLE
gfx: swipe to toggle info/watch screen

### LINKED ISSUE #55: [feature] interact with the display
I saw that we are using the lvgl as a graphic library am I right ? 

It would be nice to implement some kind of interraction with the display at some point. 

I found that page:
https://github.com/lvgl/lvgl/blob/069e24bdb294f27cc45c50497b05b5377d9f16d7/examples/porting/lv_port_indev_template.c
Unfortunately, it seems like the library is not interrupt based. It implements simply a periodic read of the x and y coordinates and make the use of that. 

We still could based the touch controller on interrupt and save the coordinate somewhere. And give those value to the read callback of the library ? 

What do you think ?

Documentation about  input devices in lvgl: 
https://docs.lvgl.io/v7/en/html/porting/indev.html?highlight=lv_indev_state_rel#

### PR DESCRIPTION
**Pull request by endian-albin**
This is a proof-of-concept that fixes #55. @sdorre, what do you think of the general approach here? Should we separate gui and gfx and put them in different modules perhaps?

### CODE REVIEW

File: app/hypnos/src/event_handler.c [hunk 1]
```diff
@@ -143,13 +148,28 @@ void touch_tap_isr(struct device *touch_dev, struct sensor_trigger *tap)
 
 	sensor_channel_get(touch_dev, CST816S_CHAN_TOUCH_POINT_1, &touch_point);
 	LOG_INF("Gesture %d on x=%d, y=%d", gesture, touch_point.val1, touch_point.val2);
+	switch (gesture) {
+	case 1:
```
Comment (sdorre) [inline app/hypnos/src/event_handler.c:152]: should use the public API for that 
`cst816s_gesture` enum is there for that

**Review by sdorre — COMMENTED** [review]
It would make sense to have some kind of Gui module here.
 More to decouple the event handler and the Screen logic. 

If we keep it separate from our Gfx module. That leads to have a pure GFX api which provides all the API to draw the different screens, elements, whatever and the GUI logic let alone to handle changing the screens and drawing the desired elements.

File: app/hypnos/src/event_handler.c (hunk 1 above)
Comment (endian-albin) [reply app/hypnos/src/event_handler.c]: sure, fixed

**Comment by endian-albin** [comment]
Yes, I think we should prepare complete screens and switch between them rather than manipulate the existing elements as done here. I also agree with your reasoning around the GUI/GFX separation but I don't yet see how it should be done. I'll need to think about it a bit more.

**Comment by sdorre** [comment]
> Yes, I think we should prepare complete screens and switch between them rather than manipulate the existing elements as done here (`gfx_show_{date|info}()` updates the label texts and styles while `gfx_update()` does all the redrawing). I also agree with your reasoning around the GUI/GFX separation but I don't yet see how it should be done. I'll need to think about it a bit more.

Simply create a gui.c which defines:
`gui_handle_touch_event()` that will be called from the tap_isr
and defines all the screen in this file and call the `gfx_show_xxx` API

**Comment by endian-albin** [comment]
Thanks for the feedback!

**Comment by endian-albin** [comment]
@sdorre, something like this?

File: app/hypnos/src/gui.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+{
+	if (sensor_sample_fetch(touch_dev) < 0) {
+		LOG_ERR("Touch sample update error.");
+	}
+	struct sensor_value gesture_val;
+	sensor_channel_get(touch_dev, CST816S_CHAN_GESTURE, &gesture_val);
+	gesture = gesture_val.val1;
+
+	sensor_channel_get(touch_dev, CST816S_CHAN_TOUCH_POINT_1, &touch_point);
+	LOG_INF("Gesture %d on x=%d, y=%d", gesture, touch_point.val1, touch_point.val2);
```
Comment (sdorre) [inline app/hypnos/src/gui.c:28]: Maybe keep the sample fetch and chaneel get in the ISR ? and give the gesture as a parameter to this function.

File: app/hypnos/include/gfx.h [hunk 3]
```diff
@@ -22,11 +22,18 @@ enum battery_symbol {
 	BAT_EMPTY
 };
 
+enum screen {
+	WATCH,
+	INFO
+};
+
```
Comment (sdorre) [inline app/hypnos/include/gfx.h:29]: that can go to gui.h as well

**Review by sdorre — COMMENTED** [review]
Look promising !

File: app/hypnos/src/gui.c (hunk 2 above)
Comment (endian-albin) [reply app/hypnos/src/gui.c]: fixed

File: app/hypnos/include/gfx.h (hunk 3 above)
Comment (endian-albin) [reply app/hypnos/include/gfx.h]: fixed

**Review by sdorre — APPROVED** [review]

**Comment by endian-albin** [comment]
@sdorre, I've now made a proper info screen containing the short commit hash and a short message and the project's URL. You can't fit that much text in one screen. I'm switching by simply hiding/showing the relevant labels.

Please try it out when convenient.

**Review by sdorre — APPROVED** [review]

**Pull request merged**